kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/format/KafkaEntryFormatter.java

```diff
@@ -111,7 +110,11 @@ public DecodeResult decode(List<Entry> entries, byte magic) {
                     }
                     optionalByteBufs.ifPresent(byteBufs -> byteBufs.add(kafkaBuffer));
                 }
-            } catch (KoPMessageMetadataNotFoundException | IOException e) { // skip failed decode entry
+                // Almost all exceptions in Kafka inherit from KafkaException and will be captured and processed in KafkaApis.
+                // Here, whether it is down-conversion or the IOException in builder.appendWithOffset in
+                // decodePulsarEntryToKafkaRecords will be caught by Kafka and the KafkaException will be thrown.
+                // So we need to catch KafkaException here.
+            } catch (KoPMessageMetadataNotFoundException | IOException | KafkaException e) { // skip failed decode entry
```

Comment:
Here, whether it is down-conversion or the IOException in builder.appendWithOffset in  decodePulsarEntryToKafkaRecords will be caught by Kafka and the KafkaException will be thrown.